insights_engine.py

```
@@ -0,0 +1,192 @@
+    """
+    if df_view.empty:
+        return ["📊 **Sin datos para el período seleccionado**\n\nAjusta los filtros para ver insights."]
+    
+    latest_year = df_view['year'].max()
+    sub = df_view[df_view['year'] == latest_year].copy()
+    
+    if sub.empty:
+        return ["📊 **Sin datos para el año más reciente**\n\nSelecciona un rango de años más amplio."]
+    
+    # Verificar si tenemos las columnas necesarias
+    required_cols = ['category', 'exp_yoy', 'balance']
```

Ruff (0.11.9)

Local variable `required_cols` is assigned to but never used

Remove assignment to unused variable `required_cols`

(F841)